keys/signer_secp256k1.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keys
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/coinbase/rosetta-sdk-go/asserter"
+	"github.com/coinbase/rosetta-sdk-go/types"
+
+	"github.com/ethereum/go-ethereum/crypto/secp256k1"
+)
+
+// SignerSecp256k1 is initialized from a keypair
+type SignerSecp256k1 struct {
+	KeyPair *KeyPair
+}
+
+// PublicKey returns the PublicKey of the signer
+func (s *SignerSecp256k1) PublicKey() *types.PublicKey {
+	return s.KeyPair.PublicKey
+}
+
+// Signs arbitrary payloads using a KeyPair
+func (s *SignerSecp256k1) Sign(payload *types.SigningPayload, sigType types.SignatureType) (*types.Signature, error) {
+	err := s.KeyPair.IsValid()
+	if err != nil {
+		return nil, err
+	}
+	privKeyBytes := s.KeyPair.PrivateKey.Bytes
+
+	if payload.SignatureType != sigType {
```

Comment:
nit: this will error if a `sigType` is passed in when `payload.SignatureType` is not populated. You should use the same logic as here: https://github.com/coinbase/rosetta-sdk-go/pull/54/files#diff-e27a53bd15dfa9595612a921778dd566R43-R45